library/Result.php

```diff
@@ -114,14 +110,25 @@ public function withIdFrom(Rule $rule): self
         return $this->clone(id: lcfirst(substr((string) strrchr($rule::class, '\\'), 1)));
     }
 
-    public function withUnchangeableId(string $id): self
+    public function withPath(string|int $path): self
     {
-        return $this->clone(id: $id, unchangeableId: true);
+        if ($this->path === $path) {
```

Comment:
Better convert the path to array and formar it as you want later

Reply:
I will do that another time, as long as it's working, this is fine now. I will make more changes to it later.